spec/support/global_state_reset.rb

```diff
@@ -0,0 +1,22 @@
+module OpenProject
+  ##
+  # Provides resets to any global state that may have been changed through tests
+  # and the change of which should not affect other tests.
+  module GlobalStateReset
+    module_function
+
+    def reset_global_state!
+      reset_current_user!
+    end
+
+    def reset_current_user!
+      User.current = User.anonymous
+    end
+  end
+end
+
+RSpec.configure do |config|
+  config.after(:suite) do
+    OpenProject::GlobalStateReset.reset_global_state!
```

Comment:
@machisuji This duplicates what we have in https://github.com/opf/openproject/blob/dev/spec/spec_helper.rb#L197. Additionally, it sets User.current to an AnonymousUser which will be left in the DB. I propose setting this to nil as well and remove the duplicity with the spec_helper.